metadata-events/mxe-utils-avro-1.7/src/main/java/com/linkedin/metadata/EventUtils.java

```diff
@@ -107,6 +113,20 @@ public static GenericRecord pegasusToAvroMCE(@Nonnull MetadataChangeEvent event)
     return renameSchemaNamespace(original, ORIGINAL_MCE_AVRO_SCHEMA, RENAMED_MCE_AVRO_SCHEMA);
   }
 
+    /**
+     * Converts a Pegasus Failed MCE into the equivalent Avro model as a {@link GenericRecord}.
+     *
+     * @param failedMetadataChangeEvent the Pegasus {@link FailedMetadataChangeEvent} model
+     * @return the Avro model with com.linkedin.pegasus2avro.mxe namesapce
+     * @throws IOException if the conversion fails
+     */
+    @Nonnull
+    public static GenericRecord pegasusToAvroFailedMCE(@Nonnull FailedMetadataChangeEvent failedMetadataChangeEvent) throws IOException {
```

Comment:
We didn't have this method in `EventUtils` because in our internal samza job we construct the Avro FMCE directly without going through Pegasus first.

Reply:
To be able to do this, generic MCE record should be first converted to specific MCE record.
Internal Samza job first converts `GenericRecord` to specific MCE avro schema using a LinkedIn proprietary util, not sure if there is a similar open source util? I think that this way should be fine and achieves the purpose anyways. What do you think @mars-lan ?

Reply:
Alternative that i could think of would be to register a custom serializer for `FailedMetadataChangeEvent.class`. Is that what you were thinking of @mars-lan

Reply:
The above alternative didn't work as it expects serializer for all the classes that FMCE uses.
Let me know in case of any alternatives?

Reply:
Thanks for the context, @arunvasudevan & @keremsahin1. In that case it's fine to add this method. Please also add the corresponding test for it.

Reply:
How are these tests invoked? 
https://github.com/linkedin/datahub/blob/master/metadata-events/mxe-utils-avro-1.7/src/test/resources/test-avro2pegasus-mce.json

Reply:
Added a test json for fmce - https://github.com/arunvasudevan/datahub/blob/enable_fmce/metadata-events/mxe-utils-avro-1.7/src/test/resources/test-pegasus2avro-fmce.json

Reply:
Hi @arunvasudevan,
Seems like we missed adding test sources for those. Just added that in my last commit: https://github.com/linkedin/datahub/commit/71f2d14503ddb810a930789764310998550e3442

Reply:
Please do a rebase and add your unit test as well.